tests/test_Producer.py

```diff
@@ -168,3 +168,26 @@ def handle_dr(err, msg):
     p.produce('mytopic', "\xc2\xc2", on_delivery=handle_dr)
 
     p.flush()
+
+
+def test_set_partioner_murmur2():
+    """
+    Test ability to set built-in partitioner type murmur
+    """
+    Producer({'partitioner': 'murmur2'})
+
+
+def test_set_partioner_murmur2_random():
+    """
+    Test ability to set built-in partitioner type murmur2_random
+    """
+    Producer({'partitioner': 'murmur2_random'})
+
+
+def test_set_invalid_partioner_murmur():
+    """
+    Assert invalid partitioner raises KafkaException
+    """
+    with pytest.raises(KafkaException) as e:
+        Producer({'partitioner': 'murmur'})
+        assert e == 'unknown builtin partitioner: murmur' in e
```

Comment:
are you certain this works, referencing the exception from with the block?
I think it needs to go in the outer block

Reply:
It would fail otherwise no? I have tested it without checking for the raised exception at all and it raised the appropriate exception

Reply:
the assert is never reached since Producer() raises an exception

Reply:
Perhaps I'm interpreting this wrong but I read it as the person raising the exception must be the last statement. Anything else should happen outside of the context. 

https://github.com/pytest-dev/pytest/blob/master/src/_pytest/python_api.py#L528-L547

Reply:
yep you are correct. 

https://github.com/pytest-dev/pytest/blob/master/src/_pytest/python_api.py#L528-L547

fixing